dotnet-client-libraries/WireMock.NET/WireMock.NET.Tests/PlanetsServiceTests.cs

```diff
@@ -0,0 +1,85 @@
+using FluentAssertions;
+using System.Net;
+using WireMock.RequestBuilders;
+using WireMock.ResponseBuilders;
+using WireMock.Server;
+using WireMock.Settings;
+using WireMockNet;
+
+namespace WireMock.NET.Tests
+{
+    public class PlanetsServiceTests : IDisposable
+    {
+        private readonly HttpClient _httpClient;
+        private readonly WireMockServer _mockServer;
+        private readonly PlanetsService _planetService;
+
+        public PlanetsServiceTests()
+        {
+            _mockServer = WireMockServer.Start(new WireMockServerSettings
+            {
+                StartAdminInterface = true
+            });
+
+            _httpClient = new HttpClient
+            {
+                BaseAddress = new Uri(_mockServer.Urls[0])
+            };
+
+            _planetService = new PlanetsService(_httpClient);
+        }
+
+        [Fact]
+        public async Task GivenThatPlanetExists_WhenGetPlanetByIdIsInvoked_ThenValidPlanetIsReturned()
+        {
+            // Arrange
+            var planet = new Planet(4, "Mars", 6779, 2, true);
+
+_mockServer
+    .Given(
+        Request.Create()
+            .UsingGet()
+            .WithPath("/planets/4"))
+    .RespondWith(
+        Response.Create()
+            .WithStatusCode(HttpStatusCode.OK)
+            .WithBodyAsJson(planet));
+
+            // Act
+            var result = await _planetService.GetPlanetByIdAsync(planet.Id);
+
+            // Assert
+            result.Should().NotBeNull();
```

Comment:
By changing Planet to be a record OR by implementing IEquatable, you can clean make these assertions clearer as this code is basically performing an equality check (with the absence of the ID property, which, since you are checking everything else and are expecting to get the planet with the given ID, this test should probably assert as well).

Reply:
Resolved in a separate PR https://github.com/CodeMazeBlog/CodeMazeGuides/pull/1182